### PR TITLE
feat(Modal|ComposedModal): add inline loading option

### PR DESCRIPTION
**Pull request by Tomahaawk**
Contributes to #10495

This PR introduces the [inline loading variant](https://carbondesignsystem.com/components/modal/usage/#loading) to Modals.  
The figma designs can be found [here](https://www.figma.com/file/uSSXI5VprCEcnuLgkfThtz/Loader-in-modal_-Spec?type=design&node-id=162-217164&mode=design).

#### Changelog

**New**

- The following properties are now available to Modal and ModalFooter: 
`loadingStatus`: loading status (active, inactive, error, finished)
`loadingDescription`: the description for the loading text 
`loadingIconDescription`: the description for the loading text 
`onLoadingSuccess`:  an optional handler to be invoked when loading is successful

- Added a new scss selector (`{prefix}--btn--loading`) that changes the button to be transparent.

#### Testing / Reviewing

- [ ] Inline loading should be active when passing `active` to `loadingStatus`
- [ ] Primary and secondary buttons should be disabled.
- [ ] Primary button background color should change to transparent when loading is active.
- [ ] After status is changed to `finished`, success state should be displayed.

### CODE REVIEW

**Comment by Tomahaawk** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by Tomahaawk** [comment]
@tay1orjones @guidari Hey, just wanted to know if you guys had time to look at the changes. 
Can the PR be set as ready for review?

File: packages/react/src/components/ComposedModal/ModalFooter.tsx [hunk 1]
```diff
@@ -251,6 +298,27 @@ ModalFooter.propTypes = {
     }),
   ]),
 
+  /**
+   * loading status
+   */
+  loadingStatus: PropTypes.oneOf(['inactive', 'active', 'finished', 'error']),
```
Comment (guidari) [inline packages/react/src/components/ComposedModal/ModalFooter.tsx]: You just have to sort alphabetically here

**Review by guidari — COMMENTED** [review]
> @tay1orjones @guidari Hey, just wanted to know if you guys had time to look at the changes. Can the PR be set as ready for review?

I few like the PR is ready for review, the functionality and code looks good to me!

**Review by guidari — APPROVED** [review]

**Review by kingtraceyj — CHANGES_REQUESTED** [review]
it looks like the loading state text is slightly off. i'm having a hard time inspecting the elements, but it looks like the text during loading is maybe 2px higher than it should be. here are the 2 states overlayed and the text  should be baseline aligned to avoid jumping:
image

**Comment by kingtraceyj** [comment]
@Tomahaawk thanks for the update! i'm still seeing that it's a little off. maybe it's the type size. are you using `$body-short-01` when it's loading?

i'm not using any tools. i guess we just do a lot of these reviews 😅

**Comment by Tomahaawk** [comment]
@kingtraceyj True, I had not used it. After adding `body-short-01` it seems the text has same height now (I hope this extension I used works properly haha)

Before adding $body-short-01
image

After adding $body-short-01
image
image

**Review by kingtraceyj — APPROVED** [review]
with the updated type token, it looks good to go! thanks for contributing 💥

**Comment by Tomahaawk** [comment]
@alisonjoseph hi, merge conflicts are solved

**Review by andreancardona — APPROVED** [review]

**Review by guidari — APPROVED** [review]
LGTM! 🚀

**Review by kingtraceyj — APPROVED** [review]

**Pull request merged**